### PR TITLE
Display single Exam and Exams list

### PR DESCRIPTION
**Pull request by bdebever**
In order to avoid any conflict, we have decided to merge @ValentinChion's work: list all exams into my branch. 

This means that the PR contains both works. 

Please review carefully, a lot of changes have been made.

### CODE REVIEW

File: backend/src/services/exam/model.js [hunk 1]
```diff
@@ -1,41 +1,51 @@
...
+        required: false,
+        trim: true,
+    },
+    reminder: {
+        type: String,
+        required: false,
+        trim: true,
+    },
+    showScale: {
+        type:Boolean,
```
Comment (NansD) [inline backend/src/services/exam/model.js:20]: A whitespace between 'type:' and 'Boolean' would look better

Comment (NansD) [reply backend/src/services/exam/model.js:20]: Also, if showscale has a default value, we can remove the 'required: false', don't you think ?

File: frontend/src/react/Exam/DisplayExam.js [hunk 2]
```diff
@@ -1,68 +1,78 @@
...
 
-    /*createExam = () => {
-        this.props.postExam({
-            title: 'This is a title from Redux',
-            'reminders': 'This is a reminder from Redux',
-            'instruction': 'These are some instructions'
-        });
-    }*/
+    closeError = () => {
+        this.setState({ displayError: false });
```
Comment (NansD) [inline frontend/src/react/Exam/DisplayExam.js:24]: Shouldn't we do :
```
this.setState({
...this.state, 
displayError: false 
});
```
?

File: frontend/src/react/Layout/Navbar.js [hunk 3]
```diff
@@ -1,27 +1,37 @@
...
+
 export default class Navbar extends Component {
+
     static propTypes = {
         title: PropTypes.string.isRequired
     };
 
     render() {
-        const { title } = this.props;
+        //const { title } = this.props;
```
Comment (NansD) [inline frontend/src/react/Layout/Navbar.js]: remove this line if it is not needed 👍

File: frontend/src/redux/exams/reducers.js [hunk 4]
```diff
@@ -29,13 +36,31 @@ export default combineReducers({
...
 
       default:
         return state;
     }
   },
+
+  errorMessage: (state = null, action) => {
+    const { type, err } = action;
+
+    if (type === ACTIONS.RESET_ERROR_MESSAGE) {
```
Comment (NansD) [inline frontend/src/redux/exams/reducers.js:63]: is it a pattern that you reproduced ?
(I have no idea on how to handle errors with redux, sorry I'm a noob)

File: frontend/src/repository/exams.js [hunk 5]
```diff
@@ -1,13 +1,46 @@
...
+    return data;
+};
+
+export const getExamsWithScaleAndTime = async () => {
+    const {data} = await axios.get('http://localhost:4000/api/exams');
+    let examScale = 0;
+    let examTime = 0;
+    for (let i = 0; i < data.length; i++) {
+        for (let j = 0; j < data[i].exercices.length; j++) {
+            for (let k = 0; k < data[i].exercices[j].questions.length; k++) {
```
Comment (NansD) [inline frontend/src/repository/exams.js:21]: triple for loop, yum 😋

File: frontend/src/style/index.scss [hunk 6]
```diff
@@ -17,10 +20,39 @@
...
+@import "helpers.scss";
+
+// TODO: find a better way of handling that through SCSS variable
 .box {
     box-shadow: 0 7px 14px 0 rgba(60,66,87, 0.1), 0 3px 6px 0 rgba(0, 0, 0, .07);
 }
 
-.pb-0 {
-    padding-bottom: 0;
+@media print {
```
Comment (NansD) [inline frontend/src/style/index.scss:30]: well done

File: frontend/src/react/Exam/DisplayExam.js (hunk 2 above)
Comment (NansD) [reply frontend/src/react/Exam/DisplayExam.js:24]: I guess this works, I think I actually used the same syntax before 🙊

File: backend/src/services/exam/model.js (hunk 1 above)
Comment (bdebever) [reply backend/src/services/exam/model.js:20]: @ValentinChion

File: frontend/src/react/Exam/DisplayExam.js (hunk 2 above)
Comment (bdebever) [reply frontend/src/react/Exam/DisplayExam.js:24]: Yep, fair point. Just didn't have anything else in my state so don't need to, do I?

Comment (NansD) [reply frontend/src/react/Exam/DisplayExam.js:24]: You don't 👍

**Review by NansD — APPROVED** [review]

File: frontend/src/repository/exams.js (hunk 5 above)
Comment (bdebever) [reply frontend/src/repository/exams.js:21]: @ValentinChion champion here! :p

Comment (bdebever) [reply frontend/src/repository/exams.js:21]: @ValentinChion , I think you can achieve that in a cleaner/easier way using reduce. 

Apply reduce and forEach. I can provide the code if needed.

File: frontend/src/redux/exams/reducers.js (hunk 4 above)
Comment (bdebever) [reply frontend/src/redux/exams/reducers.js:63]: Yes, commented on the code!

**Review by FelixDelcourt — APPROVED** [review]

**Pull request merged**